lib/zlib.js

```diff
@@ -276,6 +279,7 @@ function ZlibBase(opts, mode, handle, { flush, finishFlush, fullFlush }) {
   this._finishFlushFlag = finishFlush;
   this._defaultFullFlushFlag = fullFlush;
   this._info = opts && opts.info;
+  this._maxOutputLength = (opts && opts.maxOutputLength) || kMaxLength;
```

Comment:
Should we at least check that `this._maxOutputLength` is a number during this initialization?

Reply:
Above there's a check for 'opts` there the following code could be used:

```js
  maxOutputLength = checkRangesOrGetDefault(
    opts.maxOutputLength, 'options.maxOutputLength',
    1, Number.MAX_SAFE_INTEGER, kMaxLength
  );
```

And here:

```suggestion
  this._maxOutputLength = maxOutputLength;
```